framework/include/limiters/Limiter.h

```diff
@@ -61,23 +63,280 @@ class Limiter
+   *    reasonable range, specifically ensuring non-negative values.
+   *
+   * @note The small epsilon value `1e-10` is added to the denominator to avoid division by zero
+   *       and numerical instability.
+   *
+   * @example
+   * @code
+   * VectorValue<Real> grad_upwind(0.1, 0.2, 0.3);
+   * VectorValue<Real> grad_downwind(0.4, 0.5, 0.6);
+   * RealVectorValue dCD(1.0, 0.0, 0.0);
+   * Real ratio = rf_grad(&grad_upwind, &grad_downwind, dCD);
+   * @endcode
+   */
+  T rf_grad(const VectorValue<T> * grad_phi_upwind,
+            const VectorValue<T> * grad_phi_downwind,
+            const RealVectorValue & dCD) const
+  {
+    const auto grad_elem = (*grad_phi_upwind) * dCD;
+    const auto grad_face = (*grad_phi_downwind) * dCD;
+    const auto grad_ratio = grad_elem / (grad_face + 1e-10);
+    return std::max(2.0 * grad_ratio - 1.0, 0.0);
+  };
+
+  /**
+   * @brief Computes the flux limiting ratio using successive deltas.
+   *
+   * This function calculates the flux limiting ratio based on the upwind value,
+   * its gradient, maximum and minimum allowable values, and geometric information
+   * from the face and cell centroids.
+   *
+   * @tparam T The data type of the scalar values and the return type.
+   * @param phi_upwind The scalar value at the upwind location.
+   * @param grad_phi_upwind Pointer to the gradient vector at the upwind location.
+   * @param max_value The maximum allowable value.
+   * @param min_value The minimum allowable value.
+   * @param fi FaceInfo object containing geometric details such as face centroid and cell
+   * centroids.
+   * @param fi_elem_is_upwind Boolean indicating if the face info element is upwind.
+   * @return The computed flux limiting ratio.
+   *
+   * This function performs the following steps:
+   * 1. Retrieves the face centroid from the `FaceInfo` object.
+   * 2. Determines the cell centroid based on whether the current element is upwind.
+   * 3. Computes the delta value at the face by taking the dot product of the upwind gradient
+   *    vector with the difference between the face and cell centroids.
+   * 4. Computes the delta values for the maximum and minimum allowable values, adding a small
+   *    epsilon value to prevent division by zero.
+   * 5. Calculates the flux limiting ratio based on whether the delta value at the face is
+   *    non-negative or negative, using the appropriate delta (either max or min).
+   *
+   * @note The small epsilon value `1e-10` is added to the delta max and delta min values to
+   *       avoid division by zero and numerical instability.
+   *
+   * @example
+   * @code
+   * Real phi_upwind = 2.0;
+   * VectorValue<Real> grad_upwind(0.1, 0.2, 0.3);
+   * Real max_value = 5.0;
+   * Real min_value = 1.0;
+   * FaceInfo * fi = ... // Assume this is properly initialized
+   * bool is_upwind = true;
+   * Real ratio = rf_minmax(phi_upwind, &grad_upwind, max_value, min_value, fi, is_upwind);
+   * @endcode
```

Comment:
I was hoping we could mantain them too. Adding a utilization example is generally industry standard for clean code. Since you guys will be the main maintainers of this I am good with whatever you decide, but let's give it one more deep thought please

Reply:
Ok. So we won't document how things are used. No problem. Removed